Build error in Chamfer Distance extension: "expected initializer before ‘__s128’" in /usr/include/linux/types.h

### Description
I'm encountering a build failure when installing the project with CUDA support. The error occurs during the compilation of the CUDA extension, and the relevant error message is:

```bash
/usr/include/linux/types.h:12:27: error: expected initializer before ‘__s128’
 typedef __signed__ __int128 __s128 __attribute__((aligned(16)));
```

It appears the problem arises while compiling `chamfer.cu` using CUDA 11.1 with gcc-10. This issue seems to be related to a compatibility problem between the CUDA compiler, the version of gcc, and the Linux system headers.

### Environment
- **OS:** Ubuntu 24 under WSL (Windows Subsystem for Linux)
- **Python:** 3.7 (via conda)
- **CUDA:** 11.1
- **GCC:** gcc-10
- **PyTorch:** 1.8.0 (minimum version required for official CUDA support in WSL by Nvidia)

### Steps to Reproduce
1. Clone the repository.  
2. Create and activate a conda environment with Python 3.7.  
3. Install the required dependencies (PyTorch using `pip install torch==1.8.0+cu111 torchvision==0.9.0+cu111 torchaudio==0.8.0 -f https://download.pytorch.org/whl/torch_stable.html
`).  
4. Run the command:
   ```bash
   cd ./extensions/chamfer_dist
   python setup.py install --user
   ```
5. Observe the build process fail during the CUDA extension compilation with the error mentioned above.

--- a/README.md
+++ b/README.md
@@ -22,6 +22,27 @@ torchvision;
 pip install -r requirements.txt
 ```
 
+For Linux Kernal 6.0 or above (e.g. Ubuntu 24), please run the following command before installing Chamfer Distance:
+```
+sudo apt install gcc-10 g++-10
+
+su
+cd /usr/local/src
+wget https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.4.tar.xz
+tar -xf linux-5.4.tar.xz && cd linux-5.4
+make headers_install INSTALL_HDR_PATH=/usr/local/linux-headers-5.4
+
+export CC=/usr/bin/gcc-10
+export CXX=/usr/bin/g++-10
+export CFLAGS="-I/usr/local/linux-headers-5.4/include"
+export CPPFLAGS="-I/usr/local/linux-headers-5.4/include"
+```
+
+In `extensions/chamfer_dist/setup.py`, in the `extra_compile_args` field, pass the correct header path to nvcc by adding the following line as the second element of `ext_modules`:
+```
+extra_compile_args={"nvcc": ['--system-include=/usr/local/linux-headers-5.4/include']}
+```
+
 ```
 # Chamfer Distance & emd
 cd ./extensions/chamfer_dist

--- a/README.md
+++ b/README.md
@@ -22,7 +22,7 @@ torchvision;
 pip install -r requirements.txt
 ```
 
-For Linux Kernal 6.0 or above (e.g. Ubuntu 24), please run the following command before installing Chamfer Distance:
+For Linux Kernel 6.0 or above (e.g. Ubuntu 24), please run the following command before installing Chamfer Distance:
 ```
 sudo apt install gcc-10 g++-10
 

